src/props/onShippingAddressChange.js

```diff
@@ -95,81 +98,62 @@ export function buildXOnShippingAddressChangeActions({ data, actions: passedActi
-                    value:         `${ newAmount }`,
-                    currency_code: data?.amount?.currencyCode,
-                    breakdown
-                }
-            };
-
-            return actions;
-        },
-
-        updateShippingOptions: ({ options }) => {
-            const ordersV2Options = optionsKeyChanges(options);
-            const selectedShippingOption = options.filter(option => option.selected === true);
+        buildPatchPayload: ({tax, options, discount} = {}) => {
+            const selectedShippingOption = options?.filter(option => option.selected === true);
             const selectedShippingOptionAmount = selectedShippingOption && selectedShippingOption.length > 0
                 ? selectedShippingOption[0]?.amount?.value
                 : '0.00';
 
-            breakdown = buildBreakdown({ breakdown, updatedAmounts: { shipping: selectedShippingOptionAmount } });
-            newAmount = calculateTotalFromShippingBreakdownAmounts({ breakdown, updatedAmounts: { shipping: selectedShippingOptionAmount } });
-        
-            patchQueries[ON_SHIPPING_CHANGE_PATHS.AMOUNT] = {
-                op:       'replace',
-                path:     ON_SHIPPING_CHANGE_PATHS.AMOUNT,
-                value: {
-                    value:         `${ newAmount }`,
-                    currency_code: data?.amount?.currencyCode,
-                    breakdown
-                }
-            };
-
-            patchQueries[ON_SHIPPING_CHANGE_PATHS.OPTIONS] = {
-                op:    data?.event || 'replace', // or 'add' if there are none.
-                path:  ON_SHIPPING_CHANGE_PATHS.OPTIONS,
-                value: ordersV2Options
-            };
-
-            return actions;
-        },
-
-        updateShippingDiscount: ({ discount }) => {
-            newAmount = calculateTotalFromShippingBreakdownAmounts({ breakdown, updatedAmounts: { shipping_discount: discount } });
-            breakdown = buildBreakdown({ breakdown, updatedAmounts: { shipping_discount: discount } });
-
-            patchQueries[ON_SHIPPING_CHANGE_PATHS.AMOUNT] = {
-                op:       'replace',
-                path:     ON_SHIPPING_CHANGE_PATHS.AMOUNT,
-                value: {
-                    value:         `${ newAmount }`,
-                    currency_code: data?.amount?.currencyCode,
-                    breakdown
-                }
-            };
+            const updatedAmounts = {};
 
-            return actions;
-        },
+            if (tax) {
+                updatedAmounts.tax_total = tax;
+            }
+
+            if (selectedShippingOption) {
+                updatedAmounts.shipping = selectedShippingOptionAmount;
+            }
+
+            if (discount) {
+                updatedAmounts.shipping_discount = discount
+            }
+
+            breakdown = buildBreakdown({ breakdown, updatedAmounts });
+            newAmount = calculateTotalFromShippingBreakdownAmounts({ breakdown, updatedAmounts });
```

Comment:
I don't think we need to define `newAmount` outside of this function any longer

```suggestion
            const newAmount = calculateTotalFromShippingBreakdownAmounts({ breakdown, updatedAmounts });
```

Reply:
Word. Thanks!